SharedCode/src/commonMain/kotlin/Station.kt

```diff
@@ -0,0 +1,7 @@
+package com.jetbrains.handson.mpp.mobile
+
+class Station(val name: String, val id: String) {
+    override fun toString(): String {
```

Comment:
I believe the ArrayAdapter is using this to know what to display a Station as in the spinner.

Reply:
The `.map` method looks like a good improvement for this, although we had considered refactoring and moving the Station class functionality into a new StationModel data class that we have created for the Serialization (not part of this pull request).

I think I prefer the `.map` approach though as it seems more explicit.

Reply:
I don't like the map method, since it means when we try and receive an object from the spinner we'll get a String instead of a Station. I don't see what's wrong with overriding toString either, it's a pretty usual thing to do isn't it? It seems to me that the reason ArrayAdapter allows you to use a generic type is specifically for this type of use case.

Reply:
It is reasonable to override toString as long as you're conscious of and okay with the fact that it overrides the toString method everywhere that you're using the Station object. One thing which I think makes me slightly wary of it is that you're now tying a class that exists to contain data, which you pass around the application, to the way it is displayed in one specific part of the app. But I'm happy for the two of you to decide between you what you want to do here.